sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/aio/_client.py

```diff
@@ -172,6 +171,38 @@ async def create_ec_key(self, name: str, **kwargs: "Any") -> KeyVaultKey:
         hsm = kwargs.pop("hardware_protected", False)
         return await self.create_key(name, key_type="EC-HSM" if hsm else "EC", **kwargs)
 
+    @distributed_trace_async
+    async def create_oct_key(self, name: str, **kwargs: "Any") -> KeyVaultKey:
+        """Create a new octet sequence (symmetric) key or, if `name` is already in use, create a new version of the key.
+
+        Requires the keys/create permission.
+
+        :param str name: The name for the new key.
+        :keyword int size: the key size in bits, for example 128, 192, or 256.
+        :keyword key_operations: Allowed key operations.
+        :paramtype key_operations: list[~azure.keyvault.keys.KeyOperation or str]
+        :keyword bool hardware_protected: Whether the key should be created in a hardware security module.
```

Comment:
Since you can't create a symmetric key in a Key Vault (at least, I haven't been able to), no -- any key in an MHSM automatically gets -HSM'ed, so it'll be an oct-HSM key regardless. Maybe omitting the kwarg for now would be good?

Reply:
My thought was just that since there's technically a supported "oct" key type, we should expose this kwarg even if it's not practically useful at this time. Omitting seems fine since the general create_key method still exposes it though

Reply:
Yes, though technically AES keys are only supported in Managed HSM, which are all hardware-protected anyway. The value you respond with is ignore and all values come back as oct_HSM anyway.

Reply:
So if I understand correctly, from MHSM you get an oct-HSM key for either value of hardware_protected, while Key Vault responds with an error when you create an oct key? I lean toward omitting it then--seems like it would only be confusing.

Reply:
It's there for consistency, which is an explicit goal with the SDK (and REST APIs) for KV and MHSM. The enum values themselves have to be there as well because either is valid.

Reply:
I get that supporting both MHSM and Key Vault with the client entails adding the method even though Key Vault doesn't support it (which should be in the docstring). But why add an option that does nothing?

Reply:
Fair point, though it's in the swagger and behavior could change later. Technically, "oct" shouldn't have been there at all but GA'd in the swagger long ago.

IMO, I'd rather remain true to the swagger and not assume service behavior. That's why, for example, we don't validate non-null URL path parameters. Let the service decide. If we assume here, we're creating a tight coupling.

Reply:
This, and the fact that symmetric keys can't be created in a vault, seems to pose a bit of a documentation issue. Saying that `hardware_protected` isn't useful at this time and that symmetric keys don't work in a Key Vault seems like tight coupling to me -- but without it, the user experience is pretty bad.

Reply:
You could simply add, "This parameter may be ignored on Managed HSM". It's not a commitment but at least gives them some idea it doesn't really matter.

It's important we not tightly couple behavior to how we know something works despite how swagger is authored. Service behavior can change, but the spec (swagger) *shouldn't*.

Reply:
After some more thought, I agree with you about the API. The swagger is regrettable. I want to hide the problems here but think we should reflect the swagger because either service could change its behavior in a way that's awkward for us to accommodate if we don't.